File: mobile/lib/features/gallery/models/photo_version_type.dart
import 'package:json_annotation/json_annotation.dart';

/// Types of processed photo versions
@JsonEnum(fieldRename: FieldRename.screamingSnake)
enum PhotoVersionType {
  /// 300x300 center crop thumbnail
  thumbnail,

  /// 640px wide WebP
  webp640,

  /// 1280px wide WebP
  webp1280,

  /// 1920px wide WebP
  webp1920,

  /// 2560px wide WebP
  webp2560,
}

File: backend/src/main/java/com/rapidphoto/domain/PhotoVersionType.java
package com.rapidphoto.domain;

/**
 * Types of processed photo versions.
 */
public enum PhotoVersionType {
    THUMBNAIL,      // 300x300 center crop
    WEBP_640,       // 640px wide WebP
    WEBP_1280,      // 1280px wide WebP
    WEBP_1920,      // 1920px wide WebP
    WEBP_2560       // 2560px wide WebP
}
